### DISCUSSION TITLE
Is there an elegant way to disable QUIC support?

### DISCUSSION THREAD

**Discussion by daMatz** [mitmproxy/mitmproxy#6054] [General]
Is there an elegant way with mitmproxy to disable QUIC/HTTP3 support on Android?

I am currently testing an app which falls back to HTTP2 when UDP packages are dropped.
Therefore I want to disable QUIC / drop certain UDP packages so that I can still view the HTTP traffic.

The current way of doing this, is to intercept UDP packages via `~udp` in the mitmweb interface at the right time, but this is a little fuzzy.

It would be great to do this via an option parameter or a script to intercept only certain UDP packages and drop them.

What would be the best practice here?

**Comment by daMatz** [comment]
What I tried so far:

```
from mitmproxy import udp

def udp_message(flow: udp.UDPFlow):
    print("Server: " + flow.server_conn.address[0])
    if flow.server_conn.address[0] == 'X.X.X.X':
        print("Killing UDP flow")
        flow.kill()
```

This is giving me `mitmproxy.exceptions.ControlException: Flow is not killable.` as UDP is connectionless.

Also when trying to drop messages like in a WebSocket or TCP connectionwith `flow.messages[-1].drop()`, I can't: `AttributeError: 'UDPMessage' object has no attribute 'drop'`.

**Comment by daMatz** [comment]
Ok, so starting with `--intercept '~udp & ~dst X.X.X.X'` works for my use-case now, but would still be interested in a general solution if anybody knows anything :)

**Comment by mhils** [comment]
Yes, this is a mitmproxy 9 issue. I'd recommend upgrading to main, which both supports HTTP/3 and has a `--set http3=false` option. :)

**Reply by daMatz** [reply]
Ah nice, thanks 👍
